RecoLocalCalo/HcalRecAlgos/src/parseHFPhase1AlgoDescription.cc

```diff
@@ -30,6 +30,12 @@ parseHFPhase1AlgoDescription(const edm::ParameterSet& ps)
         const bool rejectAllFailures =
             ps.getParameter<bool>("rejectAllFailures");
 
+        float minChargeForUndershoot = -10000.f, minChargeForOvershoot = -10000.f;
+        if (ps.existsAs<double>("minChargeForUndershoot"))
```

Comment:
In a sense, the situation here is very similar to plugins. That parser function, "parseHFPhase1AlgoDescription", returns an instance of an object whose class is not known in advance. We only know that this class inherits from "AbsHFPhase1Algo" (in case of plugins, similar role is played by EDProducer, etc). This architecture allows us to decouple changes in HcalRecProducers from changes in HcalRecAlgos. It would be nice to keep it this way.